src/elements/draggable.ts

```diff
@@ -0,0 +1,85 @@
+import interact from 'interactjs';
+
+/**
+ * @param {HTMLElement} target
+ * @param {Interact.DraggableOptions} options
+ */
+export default function draggable(target, options = {}) {
+  const interaction = interact(target).draggable(Object.assign({
+    onmove: translate,
+    // keep the element within the area of it's parent
+    modifiers: [
+      interact.modifiers.restrictRect({
+        restriction: 'parent'
+      })
+    ]
+  }, options));
+  repositionOnOpen(target);
+  return interaction;
+}
+
+/**
+ * @param {Interact.InteractEvent} event
+ */
+function translate(event) {
+  const target = event.target;
+  // keep the dragged position in the data-x/data-y attributes
+  const x = (parseFloat(target.getAttribute('data-x')) || 0) + event.dx;
+  const y = (parseFloat(target.getAttribute('data-y')) || 0) + event.dy;
+  target.style.transform = `translate(${x}px, ${y}px)`;
+  // update the position attributes
+  target.setAttribute('data-x', x);
+  target.setAttribute('data-y', y);
+}
+
+function checkForOverlap(target): DOMRect[] {
+  const overlapList: DOMRect[] = [];
+  document.querySelectorAll<HTMLElement>('.ngm-floating-window').forEach(elem => {
+    if (!elem.hidden && elem !== target) {
+      const targetRect = target.getBoundingClientRect();
+      const checkRect = elem.getBoundingClientRect();
+      const overlap = !(targetRect.right < checkRect.left || targetRect.left > checkRect.right);
+      if (overlap) overlapList.push(checkRect);
+    }
+  });
+  return overlapList;
+}
+
+function moveWindow(target, overlapList, changeSide) {
```

Comment:
I don't understand all the logics here. Could you add some comments?

I would expect to simply try positioning at top: 0, right: XX . Then if there is collision move the target to the left of the colliding window, check collision and move again if needed. In the case where we are too far on the left (window would go out of the the viewport), then I would put the window at right: XX.
XX: the left + margin of the +0- bar.
We should also not overlap the panels. If there is an overlap => top: 0, right: XX.

That said I am not against a more clever logics.

Reply:
I added comments.

As I said it works not perfectly now but I already spent 4h that issue was estimated (+ it is an investigation issue) so I am not sure that I should continue making updates